_content/tour/ukr/composition-pollution.article

```diff
@@ -1,86 +1,86 @@
+або коли це доцільно. У використанні інтерфейсів є своя ціна: певний рівень опосередкованості
+та додаткове виділення памʼяті, коли ми зберігаємо значення всередині. Якщо ціна додаткового виділення памʼяті
+не переважає того, що ми отримуємо цим опосередкуванням, то використання інтерфейсів треба уникати.
 
-"I need to be able to test my code so I need to use interfaces".
+"Я маю тестувати свій код, тому мені треба використовувати інтерфейси".
 
-No. you must design my API for the user first, not my test. If the API is not testable,
-You should question if it’s usable. There are different layers of API’s as well. The
-lower level unexported API’s can and should focus on testability. The higher level
-exported API’s need to focus on usability.
+Ні. Ви маєте створювати свій API в першу чергу для користувача, а не для тестів. Якщо API неможливо відтестувати,
+вам треба задатися питанням, чи їм можна користуватися. Також є різні рівні API.
+Низькорівневі, не експортовані назовні API, можуть і мають фокусуватися на зручності тестування.
+Високорівневі, експортовані назовні API, мають фокусуватися на зручності використання.
 
-Functions that accept raw data in and return raw data out are the most testable.
-Separate the data transformation from where the data comes from and where it is
-going. This is a refactoring exercise you need to perform during the engineering
-coding cycle.
+Функції, які приймають сирі дані всередину і повертають сирі дані назовні легше за все тестувати.
+Відокремлюйте процес трансформації даних від того звідки дані приходять і куди
+відправляються. Це та вправа з рефакторингу, яку вам треба робити під час інженерного
+циклу розробки.
 
-Below is an example that creates interface pollution by improperly using an
-interface when one is not needed.
+Нижче можна побачити приклад, який створює забруднення інтерфейсами, використовуючи неналежним чином інтерфейс,
+коли він не потрібен.
 
     type Server interface {
         Start() error
         Stop() error
         Wait() error
     }
 
-The Server interface defines a contract for TCP servers. The problem here is I
-don’t need a contract, you need an implementation. There will only be one implementation
-as well, especially since you are the one implementing it. You do not need someone else
-to implement this for you.
+Інтерфейс Server визначає контракт для TCP серверу. Проблема тут полягає у тому, що
+вам не потрібен контракт, вам потрібна імплементація. Буде існувати лише одна імплементація,
+особливо якщо ви саме той/та хто буде її імплементувати. Вам не потрібно, щоб хтось інший
+створював вам цю імплементацію.
 
-Plus, this interface is based on a noun and not a verb. Concrete types are nouns
-since they represent the concrete problem. Interfaces describe the behavior and
-Server is not behavior.
+До того ж, цей інтерфейс заснований на іменнику а не на дієслові. Конкретні типи це іменники,
```

Comment:
```suggestion
До того ж, цей інтерфейс ґрунтується на іменнику а не на дієслові. Конкретні типи це іменники,
```

https://onlinecorrector.com.ua/uk/%D1%89%D0%BE-%D2%91%D1%80%D1%83%D0%BD%D1%82%D1%83%D1%94%D1%82%D1%8C%D1%81%D1%8F-%D0%BD%D0%B0